### PR TITLE
address multiple timestamp issues with multi

### PR DESCRIPTION
**Pull request by notimaginative**
A bit overkill for some of this but just covering the bases.

 * fix issues with in-game join
 * fix multi messaging and multi pause screen
 * make sure maintenance stuff (pings, voice, file xfer, etc.) isn't
   affected by pausing or time compression
 * fix sync issues caused by options menu and screenshots

### CODE REVIEW

**Comment by notimaginative** [comment]
@Goober5000 Please check me on this when you get a chance. Some of it probably isn't needed in the general UI but I preferred to go overkill rather than miss something.

File: code/network/multi_ingame.cpp
Comment (Goober5000) [inline code/network/multi_ingame.cpp:953]: I recommend using `MILLISECONDS_PER_SECOND` here.

File: code/network/multi_ingame.cpp
Comment (Goober5000) [inline code/network/multi_ingame.cpp:477]: I recommend setting this to just 15 and then using `MILLISECONDS_PER_SECOND` when the actual timestamp is created.

File: code/network/multi.cpp
Comment (Goober5000) [inline code/network/multi.cpp:1302]: Might want to use `MILLISECONDS_PER_SECOND` here.

File: code/network/multi.cpp
Comment (Goober5000) [inline code/network/multi.cpp:1402]: Here too

File: code/network/multi_observer.cpp
Comment (Goober5000) [inline code/network/multi_observer.cpp:63]: Looks like you can convert this to use `TIMESTAMP`, and use `TIMESTAMP::immediate()`

File: code/network/multi_observer.cpp
Comment (Goober5000) [inline code/network/multi_observer.cpp:59]: Looks like `last_heard_time` could be converted to `TIMESTAMP` or `UI_TIMESTAMP`

File: code/network/multi_observer.cpp
Comment (Goober5000) [inline code/network/multi_observer.cpp:61]: I'd say `kick_timestamp` could be converted too

File: code/network/multiutil.cpp
Comment (Goober5000) [inline code/network/multiutil.cpp:806]: Same comments re `kick_timestamp` and `player_collision_timestamp`

File: code/network/multiutil.cpp
Comment (Goober5000) [inline code/network/multiutil.cpp:1728]: This function looks like a candidate for smart pointers, but that's out of scope for this PR.

File: code/network/multi_pxo.cpp
Comment (Goober5000) [inline code/network/multi_pxo.cpp:563]: I'd suggest making this `2 * MILLISECONDS_PER_SECOND`

File: code/network/multi_pxo.cpp
Comment (Goober5000) [inline code/network/multi_pxo.cpp:1019]: I'd suggest making this `120 * MILLISECONDS_PER_SECOND`

File: code/network/multi_voice.cpp
Comment (Goober5000) [inline code/network/multi_voice.cpp:104]: Suggest `7 * MILLISECONDS_PER_SECOND`

File: code/network/multi_voice.cpp
Comment (Goober5000) [inline code/network/multi_voice.cpp:148]: Suggest `1 * MILLISECONDS_PER_SECOND`

File: code/network/multi_voice.cpp
Comment (Goober5000) [inline code/network/multi_voice.cpp:2079]: I don't think the logical NOT is correct here.

**Review by Goober5000 — COMMENTED** [review]
I like this; these are good and sensible changes.  I don't think it's overkill at all.  I had several comments.

(The `MILLISECONDS_PER_SECOND` suggestions are optional; they might improve clarity but they might also get in the way.  Your call.  And I'm sure there are others I didn't catch.)

File: code/network/multi_voice.cpp
Comment (notimaginative) [reply code/network/multi_voice.cpp:2079]: Yep, I got that wrong in a few places but must have missed this one when going back to fix them.

**Comment by notimaginative** [comment]
I think that covers the required changes. I make a couple of the `MILLISECONDS_PER_SECOND` changes, but I'd prefer to hold off on the rest for a proper cleanup PR and leave this one to focus on the real fixes.

File: code/network/multi_observer.cpp
Comment (notimaginative) [reply code/network/multi_observer.cpp]: Done

File: code/network/multi.cpp
Comment (notimaginative) [reply code/network/multi.cpp]: Fixed

File: code/network/multi.cpp
Comment (notimaginative) [reply code/network/multi.cpp]: Fixed

File: code/network/multi_observer.cpp
Comment (Goober5000) [reply code/network/multi_observer.cpp:59]: Ok.

**Review by Goober5000 — APPROVED** [review]

**Pull request merged**
